driver/bpf/filler_helpers.h

```diff
@@ -273,16 +282,17 @@ static __always_inline bool bpf_getsockname(struct socket *sock,
 static __always_inline int bpf_addr_to_kernel(void *uaddr, int ulen,
 					      struct sockaddr *kaddr)
 {
-	int len = _READ(ulen);
+	int len = ulen & 0xfff;	/* required by BPF verifier */
```

Comment:
> I'm confused here, why did we read an `int` with the `bpf_probe_read` helper before your patch? fearful

Honestly, I am not sure. I looked up the original commit for this and for me it seemed that the context where it was used has changed. The issue I faced with the `_READ` that is returned just `0` because `ulen` is already read / passed as register before calling this help

Reply:
yeah is completely a no-sense :thinking: probably we need to rewrite the whole helper one of these days